jsonschema-core.md

```diff
@@ -523,11 +523,23 @@ to inform the user of the evaluation path that produced the error or annotation.
 
 ### Keyword Interactions
 
-Keyword behavior MAY be defined in terms of the annotation results of
-[subschemas](#root) and/or adjacent keywords (keywords within the same schema
-object) and their subschemas. Such keywords MUST NOT result in a circular
-dependency. Keywords MAY modify their behavior based on the presence or absence
-of another keyword in the same [schema object](#schema-document).
+Keywords MAY modify their behavior based on the presence, absence, or value of
+another keyword in the same [schema object](#schema-document). Such keywords
+MUST NOT result in a circular dependency.
+
+Unless otherwise specified, keywords act independently.
+
+#### Annotations as an Interaction Mechanism
+
+This document uses annotations as a mechanism to describe the interactions
+between keywords, however this is for illustrative purposes and not
+prescriptive. Implementations MAY use whatever mechanisms makes sense given
+their architecture and language needs in order to achieve the specified
+behaviors. Implementations are not required to collect annotations in order to
+manage keyword interaction.
+
+Supplementary specifications are also encouraged to use annotations to describe
```

Comment:
I would delete this line personally. I don't know that I believe that it's anything other than a stylistic choice to define keywords in this way, and I'm not really convinced it's clearer to do so in general -- worse so given we ourselves have had unexpected issues from having done so (c.f. https://github.com/orgs/json-schema-org/discussions/57)

I'm going to LGTM the PR anyhow, but yeah, I would delete this personally.

Reply:
I agree, that's why I didn't use SHOULD or make it any kind of hard requirement.  I think it's good for related specs to be aligned in their language.  It's just a gentle nudge.

Reply:
> I would delete this line personally.

I would too. I too don't think describing behaviors using annotations is more clear, but I'm more concerned that this line (as well as describing keyword behaviors in terms of annotations in general) will (indirectly) encourage people to implement things in a specific way. I personally don't think encouraging stateful evaluation through annotations is a good choice because it can lead to less efficient implementations.

So, while this PR isn't exactly what I would want, I think it's a significant step in the right direction and I'll take what I can get.